networkdb/delegate.go

```diff
@@ -63,10 +82,27 @@ func (nDB *NetworkDB) purgeSameNode(n *node) {
 }
 
 func (nDB *NetworkDB) handleNodeEvent(nEvent *NodeEvent) bool {
-	n := nDB.checkAndGetNode(nEvent)
+	// Update our local clock if the received messages has newer
+	// time.
+	nDB.networkClock.Witness(nEvent.LTime)
+
+	n := nDB.getNode(nEvent)
 	if n == nil {
 		return false
 	}
+	// If its a node leave event for a manager we want the reconnect logic to
+	// kick in. Because in a single manager cluster manager's gossip can't be
+	// bootstrapped unless some other node connects to it.
+	if nEvent.Type == NodeEventTypeLeave {
+		for _, ip := range nDB.bootStrapIP {
```

Comment:
I was considering making it specific to the one manager case. But in the multiple manager case also workers connecting to a restarting manager should be ok, isn't it ? It will help in that manager node converging faster.